libraries/botbuilder-dialogs-adaptive/src/selectors/conditionalSelector.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Expression, ExpressionEngine } from "botframework-expressions";
+import { OnCondition } from "../conditions/onCondition";
+import { TriggerSelector } from "../triggerSelector";
+import { SequenceContext } from "../sequenceContext";
+
+/**
+ * Select between two rule selectors based on a condition.
+ */
+export class ConditionalSelector implements TriggerSelector {
+    private _conditionals: OnCondition[];
+    private _evaluate: boolean;
+    private _condition: Expression;
+
+    /**
+     * Gets or sets expression that determines which selector to use.
+     */
+    public get condition(): string { return this._condition.toString() }
+    public set condition(value: string) { this._condition = (value != null) ? new ExpressionEngine().parse(value) : null }
```

Comment:
>(value != null) ? [](start = 60, length = 17)

value ?

That will ensure that value is not (null || undefined) and has a length.